### ISSUE TITLE
Checker bytecode hashes should be generated from source code

### ISSUE THREAD

**Issue by yorhodes** [hyperlane-xyz/hyperlane-monorepo#1961]
The currently static bytecode hashes in the checker https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/44b26ff70189861bd2948336c40735b63038bb6c/typescript/sdk/src/consts/bytecode.ts should be retrieved from `factory.bytecode`
https://docs.ethers.org/v5/api/contract/contract-factory/#ContractFactory--properties
we may need to use `getDeployTransaction` for immutables which affect the bytecode
https://docs.ethers.org/v5/api/contract/contract-factory/#ContractFactory--methods

**Comment by yorhodes** [comment]
all of these uses should be updated as well https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/44b26ff70189861bd2948336c40735b63038bb6c/typescript/sdk/src/core/HyperlaneCoreChecker.ts#L93

**Comment by nambrot** [comment]
Do we need #1927 before this? Otherwise i can see it anoying that changing source code rn causes the checker to complain

**Comment by yorhodes** [comment]
> Do we need #1927 before this? Otherwise i can see it anoying that changing source code rn causes the checker to complain

imo its a good thing that checker complains
deployer & govern should also resolve

**Comment by nambrot** [comment]
yeah i would wanna understand how common deploy flows happen first before causing source code changes in bytecode to force remedy rn

**Comment by asaj** [comment]
> > Do we need #1927 before this? Otherwise i can see it anoying that changing source code rn causes the checker to complain
> 
> imo its a good thing that checker complains deployer & govern should also resolve but maybe that is the resolution to #1927

I like this
